api/src/main/java/com/vimeo/networking2/utils/CallExtensions.kt

```diff
@@ -0,0 +1,29 @@
+package com.vimeo.networking2.utils
+
+import kotlinx.coroutines.suspendCancellableCoroutine
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+
+/**
+ * This is an extensions on a call to make an API in a suspending
+ * function. When the request comes back, the method will resume
+ * the suspending method.
+ */
+suspend fun <T : Any> Call<T>.awaitResponse(): Response<T> =
```

Comment:
since this requires coroutines, should it be in an extension module (we can come up with a name for the module) rather than the main api module